### ISSUE TITLE
"expect" rule: newest forEachChild cannot traverse ASTs from TS <= 2.9

### ISSUE THREAD

**Issue by cspotcode** [Microsoft/dtslint#119]
dtslint's "expect" rule uses `ts.forEachChild` from "typescript@next", but the AST is from an earlier version of TypeScript.  The newest `forEachChild` implementation does not visit any children of older ASTs; it looks like something changed for TS3.0.  This breaks the "expect" rule since it cannot visit any nodes, so none of the assertions are evaluated.

I'm using dtslint's "expect" rule in my project's tslint configuration, as explained here: https://github.com/andy-ms/dtslint-ts-user

## Reproduction

```typescript
import * as ts29 from './ts2.9/node_modules/typescript'; // TS@2.9.1
import * as tsnext from './tsnext/node_modules/typescript'; // TS@next (3.0.0-dev.20180609)

console.log(ts29.version);
console.log(tsnext.version);

// Parse with typescript@2.9.1
const sourceFile = ts29.createSourceFile('foo.ts', `
    codeHere;
`, ts29.ScriptTarget.Latest, true, ts29.ScriptKind.TS);

// Traverse via typescript@next
tsnext.forEachChild(sourceFile, function(node) {
    console.log('this message is never logged');
});
```

### Output
```
2.9.1
3.0.0-dev.20180609
```

**Comment by eps1lon** [comment]
The problem is that `dtslint` lists typescript as a dependency while `tslint` only requires it as a peerDependency. This way tslint and dtslint can become out of sync and can report all sorts of false-positives (e.g. `void-return`, `strict-export-declare-modifiers` etc.).

What should be happening is that `dtslint` only requires `tslint` as a dependency  and list `typescript` as a peer and as devDependency. That way we could ensure that the AST is not passed between typescript version. 

As a workaround you can delete the `typescript` folder in the `node_modules` of `dtslint`. Just remember that this will be reverted on every npm/yarn install.

It should not be pinned to `next` to begin with IMO.

**Comment by andy-ms** [comment]
`forEachChild` was added as a method of node in #14581. If we changed the minimum TypeScript version to 2.4 we could just use the method version instead of `ts.forEachChild`.
